### DISCUSSION TITLE
GET parameter retrieval and Unicode decoding

### DISCUSSION THREAD

**Discussion by drc-ksafin** [micropython/micropython#13444] [ESP32] [answered]
Hi all!

I'm using the micropython captive portal made by metachris (https://github.com/metachris/micropython-captiveportal) to have the user connect to the ESP32 and provide credentials for their WiFi network.

It's a bit out of my depth, but I fumbled through some edits to extract the inputs from an HTML form I made, as such:
![image](https://github.com/micropython/micropython/assets/153777445/8f15c030-b714-4062-9525-dc2b9e07e6f2)

Note that I simply noticed the format of the incoming GET request with the form data and parsed the field information from this string.

I'm having one problem here though, which is that symbols are being encoded as unicode in the convention of "%xx" where xx is the unicode ID. I was starting to write a manual regex method to conduct substitutions, but was wondering if there's a better way of doing this.

I want to avoid the manual route if there's something comprehensive, because I will inevitably omit some character(s) and then funky passwords or SSID's will be non-functional.

Thanks!

**Comment by glenn20** [comment, marked as answer]
Its not unicode encoding, its `application/x-www-form-urlencoded` url encoding. On python, you usually decode that with `urllib.parse.unquote()`, which I believe is not supported by micropython on most ports. There is a compact solution for `unquote()` available on the old forum <https://forum.micropython.org/viewtopic.php?t=3076>. See the last message on that page.

**Reply by drc-ksafin** [reply]
Thanks you! The solution for unquote() worked great. Appreciate the help!

**Comment by sophiedegran** [comment]
The best thing to do is to replace with javascript unicode char before posting html ...
